test/e2e/helm/kuma_helm_deploy_global_universal_mode_and_zone.go

```diff
@@ -0,0 +1,160 @@
+package helm
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	http_helper "github.com/gruntwork-io/terratest/modules/http-helper"
+	"github.com/gruntwork-io/terratest/modules/random"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+
+	api_server "github.com/kumahq/kuma/pkg/api-server"
+	"github.com/kumahq/kuma/pkg/config/core"
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/deployments/postgres"
+	"github.com/kumahq/kuma/test/framework/deployments/testserver"
+)
+
+func ZoneAndGlobalInUniversalModeWithHelmChart() {
+	var clusters Clusters
+	var c1, c2 Cluster
+	var global, zone ControlPlane
+
+	BeforeAll(func() {
+		var err error
+		clusters, err = NewK8sClusters(
+			[]string{Kuma1, Kuma2},
+			Silent)
+		Expect(err).ToNot(HaveOccurred())
+
+		c1 = clusters.GetCluster(Kuma1).
+			WithTimeout(6 * time.Second).
+			WithRetries(60)
+		c2 = clusters.GetCluster(Kuma2).
+			WithTimeout(6 * time.Second).
+			WithRetries(60)
+
+		releaseName := fmt.Sprintf(
+			"kuma-%s",
+			strings.ToLower(random.UniqueId()),
+		)
+
+		err = NewClusterSetup().
+			Install(Namespace("kuma-system")).
+			Install(postgres.Install(Kuma1,
+				postgres.WithK8sNamespace("kuma-system"),
+				postgres.WithUsername("mesh"),
+				postgres.WithPassword("mesh"),
+				postgres.WithDatabase("mesh"),
+				postgres.WithPrimaryName("postgres"),
+				// we deploy postgres in "kuma-system" so it's deleted with CP teardown
+				// otherwise this fails because there is no namespace
+				postgres.WithSkipNamespaceCleanup(true),
+			)).
+			// Install(WaitService("kuma-system", "postgres-release-postgresql")). // this does not seem to work
+			// control plane crashes twice waiting for postgres to come up
+			Install(YamlK8s(`
+apiVersion: v1
+kind: Secret
+metadata:
+  name: postgres
+  namespace: kuma-system
+type: Opaque
+stringData:
+  password: "mesh"
+`)).
+			Setup(c1)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = NewClusterSetup().
+			Install(Kuma(core.Global,
+				WithSkipDefaultMesh(true),
```

Comment:
I'd delete it

Reply:
Without this the `DeployKuma` will fail because on k8s we expect it to create a default mesh https://github.com/kumahq/kuma/blob/c2f131ad88b3d38ec01b2a89f2ee2b1f33760127/test/framework/k8s_cluster.go#L606

Reply:
but Kuma without this flag would create a default Mesh. We do have retries if mesh exists before we consider Kuma deployed.

Reply:
Please add a comment that it's required because we check if Kuma is ready because we use `kubectl get mesh` which is not available in universal mode.

Reply:
done